### ISSUE TITLE
Validate file names when saving.

### ISSUE THREAD

**Issue by matugol** [SwissAndOr/Advanced-Graphing-Calculator#13]
Java doesn't validate file names by default when using JFileChooser to save files, and so we'll have to implement that ourselves. Some invalid file names include:
- Anything with these characters: `\ / * ? " < > | :`
- Unicode symbols
- Some combinations of periods, such as `.` or `..`
- com[1-9]
- lpt[1-9]
- con
- aux
- nul
- prn
- Whitespace (except spaces?)
- Any other file names that I might not know about.

**Comment by etillison3350** [comment]
We should just catch exceptions rather than trying to test all of these

**Comment by matugol** [comment]
Yeah that's what I said but then you said that you wanted to make them untypable/savable.
